### ISSUE TITLE
Master Child capability for Clarity Datagrid

### ISSUE THREAD

**Issue by vkpavan26** [vmware/clarity#403]
**Select one ...**  (check one with "x")
```
[ ] bug
[x ] feature request
[ ] enhancement
```

### Expected behavior
As part of our usecases we have Datagrids where UI represents Master/Detail sort of record in Datagrid using Expand/Collapse option.

When master records are expanded usually we give backend call to fetch all the required information and add it as child record.

This sort of paradigm is used in multiple usecases in our existing UI and it is a priority item for us.
Attaching few references of UI screenshots:

![image](https://cloud.githubusercontent.com/assets/25317393/22239850/4d247344-e23e-11e6-9ae1-89252a0b16b9.png)

![image](https://cloud.githubusercontent.com/assets/25317393/22239906/7e938f14-e23e-11e6-9a24-926ad39d3380.png)

**Comment by halachliyskid** [comment]
@reddolan @jaffoneh  I'm guessing it's a bit early to ask the question, but roughly when should we expect for this to be done (not only designed, but also included in Clarity)?

**Comment by reddolan** [comment]
@halachliyskid design has started on it. This generally takes a couple weeks. @jaffoneh can give you a better estimate with development included

**Comment by mathisscott** [comment]
@halachliyskid and @reddolan : It is still a ways off, development-wise. My understanding is the datagrid will need to be refactored to accommodate master-child views. After design is completed, we are still looking at a non-trivial implementation effort. It won't be constructive to throw out dates at this point in time because the feature is not fully defined and the fluidity of our priorities oftentimes can negate SWAG estimates in real time.

TL;DR It's going to be a while. End of year while? Probably not. But we're probably looking at months away.

**Comment by halachliyskid** [comment]
Thanks for the quick reply guys!

**Comment by mathisscott** [comment]
You're welcome, @halachliyskid. Re-reading my reply, I see I may have not used the best wording for where we have this item prioritized. This is one of our 3-4 top priorities. So when I said a "while", I'm afraid it came off as sounding like a long time off. What is more accurate to say is that this is one of the things we are trying to get into either 0.9 or the minor version after it (0.10).

We are currently on 0.8. I will try not to be so vague in the future.

**Comment by reddolan** [comment]
We have merged your request into Grid Row Detail #547. Please read the documentation and subscribe for updates. Comments and questions can be posted there.

Thank you
